src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php

```diff
@@ -129,13 +129,13 @@ private function addTwigOptions(ArrayNodeDefinition $rootNode)
                 ->end()
                 ->scalarNode('autoescape_service')->defaultNull()->end()
                 ->scalarNode('autoescape_service_method')->defaultNull()->end()
-                ->scalarNode('base_template_class')->example('Twig_Template')->end()
+                ->scalarNode('base_template_class')->example('Twig_Template')->cannotBeEmpty()->end()
                 ->scalarNode('cache')->defaultValue('%kernel.cache_dir%/twig')->end()
                 ->scalarNode('charset')->defaultValue('%kernel.charset%')->end()
-                ->scalarNode('debug')->defaultValue('%kernel.debug%')->end()
-                ->scalarNode('strict_variables')->end()
+                ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
+                ->booleanNode('strict_variables')->end()
                 ->scalarNode('auto_reload')->end()
-                ->scalarNode('optimizations')->end()
+                ->integerNode('optimizations')->end()
```

Comment:
optimizations are a binary flag. So it cannot be an enum node, unless we have all combinations of the binary flag (and we would have to update it each time Twig adds a new optimization)